### DISCUSSION TITLE
Debian Linux Package

### DISCUSSION THREAD

**Discussion by trymeouteh** [zen-browser/desktop#1593] [Ideas]
Please offer a bare metal DEB installer which will install Zen and add a Zen APT repository for future updates.

**Comment by ghost** [comment]
That's the only reason I haven't tried Zen yet. I've been waiting for months, maybe one day we'll get there ;-)

**Reply by KarimElsayad247** [reply]
Just a curious question, why is Flatpak not suitable for you?

**Reply by ghost** [reply]
@KarimElsayad247 sandbox so it breaks e.g. passwords managers.

I'm happy with Floorp, that's why I don't do anything about it. If there will be repo, maybe I try Zen. If not, I don't need it anyway.

**Comment by nocanoa** [comment]
If anyone want to contribute and make a workflow that does the .dep package u welcome,
Its hard to maintain/create it for 3 OSs and file types.

**Reply by homeschoolu** [reply]
https://github.com/GitXpresso/Browsers-NoVNC/releases/download/TarAndDeb/zen.linux-x86_64.deb

**Reply by homeschoolu** [reply]
but ...oops ...download and install ...nothing ??
Does anybody know why?

**Comment by senntore** [comment]
I'm also hoping for debian packaging for faster startup.
Speech dispatcher doesn't work with the flatpak package. It works with appimage & the archive release but both of them starts somewhat slow, especially the appimage.

**Comment by schneider-technology** [comment]
https://github.com/schneider-technology/zen-browser-deb

I wrote a script which creates a deb package out of the tarball - please note that this is in really early stages and my first Debian package ever. So use it at your own risk. Currently, I've only used it in a VM. I have to clean it up slightly so uninstalling works cleanly (see readme). Also I didn't test updating yet.

Please open an issue/pull request and I will try to improve the script.

**Reply by schneider-technology** [reply]
Just gonna bump this since it should be usable now. You may disagree with placing all the data in /usr/lib/zen-browser, but that's how firefox-esr does it and therefore I simply did the same. If anyone has more experience than I with such things, please create an issue or pull request.

I hope the team considers offering debian packages, but for now I will just build them myself.

**Comment by sh4r10** [comment]
I've also written a script which basically reverse engineers the provided install script to create a deb package from the linux tarball, as opposed to the Appimage. I'm using the built debs myself on my main daily driver debian system and have not encountered any issues so far. It is also my first deb package so be nice, and if you don't like something you are free to change it in the script config to suite your liking. I've tried to document it as well as possible so that it can be used with as little friction as possible. I also build the deb packages and try to keep consistent with the release of new updates if that works better for you. Here is the repo: [https://github.com/sh4r10/zen-browser-debian](https://github.com/sh4r10/zen-browser-debian).

**Comment by KaKi87** [comment]
Hi,

Some time ago, I started developing a tool named [dynapt](https://git.kaki87.net/KaKi87/dynapt) to automatically fetch updates through APT for packages that don't provide an APT repository.

As it's still experimental, you can download the latest pre-release of it and set it up from [this tutorial](https://gist.github.com/KaKi87/f620788e9901abbfef4978eb7ad358b4).

For Zen specifically, replace the following configuration line :

```json
    "apps": []
```

With the following :

```json
    "apps": [
        {
            "name": "Zen Browser",
            "github": {
                "repo": "zen-browser/desktop",
                "filter": "zen-x86_64.AppImage"
            },
            "appimage": true
        }
    ]
```

Replace `x86_64` with `aarch64` if applicable.

Since it's still AppImage-based, it's not going to beat the AppImage itself in terms of performance and features (though it still won't have Flapak's drawbacks) but at least you'll never be out of date.

Enjoy !

**Comment by Kevinhykuo** [comment]
Literally the only thing holding back Zen on Kubuntu for me. Flatpak is not compatible with the Plasma-Browser-Integration extension due to sandboxing. Having better integration with KDE and being able to have it run would be great.

**Reply by MurderFromMars** [reply]
> Literally the only thing holding back Zen on Kubuntu for me. Flatpak is not compatible with the Plasma-Browser-Integration extension due to sandboxing. Having better integration with KDE and being able to have it run would be great.

I built a bash.scriot.rk automatically fetch the latest Linux rarball and package it into a deb file. 
https://github.com/MurderFromMars/Zen-Browser-Deb-Builder

**Reply by KaKi87** [reply]
That's basically the same thing dynapt does

EDIT: nevermind, it packages the `AppImage` as a DEB, but still, it comes as an APT repository, so it updates automatically when running `apt update`.

**Reply by MurderFromMars** [reply]
> That's basically the same thing dynapt does
> 
> EDIT: nevermind, it packages the `AppImage` as a DEB, but still, it comes as an APT repository, so it updates automatically when running `apt update`.

welp mine was primarily developed for PikaOS which recommends against adding 3rd party repositories where possible

**Reply by KaKi87** [reply]
Well, for automated updates, *where possible* doesn't apply here, unless they include Zen in their official repository.

**Reply by MurderFromMars** [reply]
> Well, for automated updates, _where possible_ doesn't apply here, unless they include Zen in their official repository.

well precisely my point they don't.  so i came up with a script to build a deb 

i've overhauled the script and added a service that checks for updates daily automatically. if one is detected it will automatically update the browser from the new latest tarball. no APT repo required.

**Reply by KaKi87** [reply]
> well precisely my point they don't. so i came up with a script to build a deb

Well, my point was : since *where possible* doesn't apply, an APT repository is relevant here.

> plus with zens habits of... forcing new featurres on people in updates manually updating isn't so bad.

I meant *update* in the context of APT, as in `apt update` : the update is automatically fetched, and then you can manually apply it using `apt upgrade` (after reviewing the changelog if you want), or indeed you can apply it automatically if you configure your distro to.

The point of dynapt is to interface with APT so that your updating habits, whatever they are, keep working :

- You do `apt update` and `apt upgrade` in the terminal manually ? It works.
- You run `apt update && apt upgrade -y` in a cron ? It works.
- You use GNOME Apps or KDE Discover ? It works.
- You use the *Apply updates automatically* option on Linux Mint ? It works.

Etc.

> i've overhauled the script and added a service that checks for updates daily automatically. if one is detected it will automatically update the browser from the new latest tarball. no APT repo required.

But again, that's disconnected from people's usual update management habits.

**Reply by MurderFromMars** [reply]
welp short of zen browser devs getting off their ass and packaging their project themselves i think it's the best solution.

install and forget
